tests/app/test_model.py

```diff
@@ -1034,3 +1038,83 @@ def test_work_order_can_be_serialized(self):
                         'supplier': (('.get_supplier',), {'code': 0}),
                     }
                 }
+
+
+class TestApplication(BaseApplicationTest):
```

Comment:
What about a test using the API to set status to submitted and then ensuring it is not saved?